File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TableViews_ViewModel.cpp
#include "TableViews_ViewModel.h"
#include <vector>
#include <memory>
#include "TableViews_ViewModelMyElementsRow.h"
#include "TableViews_ViewModelMyStringRowHandleRow.h"
#include "TableViews_ViewModelMyStringMultiRowHandlesRow.h"
#include "TableViews_ViewModelFirstColumnRowHandleRow.h"
#include "TableViews_ViewModelFirstColumnMultiRowHandleRow.h"
#include <string>

namespace widgetassertions
{
  std::vector<std::shared_ptr<TableViews_ViewModelMyElementsRow>> TableViews_ViewModel::getMyElementsTableRows()
  {
    return this->MyElementsTableRows;
  }
  int TableViews_ViewModel::getMyElementsTableSelectedRow()
  {
    return this->MyElementsTableSelectedRow;
  }
  bool TableViews_ViewModel::getIsMyElementsTableVisible()
  {
    return this->isMyElementsTableVisible;
  }
  bool TableViews_ViewModel::getIsMyElementsTableEnabled()
  {
    return this->isMyElementsTableEnabled;
  }
  bool TableViews_ViewModel::getIsMyElementsTableDescriptionColumnVisible()
  {
    return this->isMyElementsTableDescriptionColumnVisible;
  }
  std::vector<std::shared_ptr<TableViews_ViewModelMyStringRowHandleRow>> TableViews_ViewModel::getMyStringRowHandleTableRows()
  {
    return this->MyStringRowHandleTableRows;
  }
  std::string TableViews_ViewModel::getMyStringRowHandleTableSelectedRow()
  {
    return this->MyStringRowHandleTableSelectedRow;
  }
  std::vector<std::shared_ptr<TableViews_ViewModelMyStringMultiRowHandlesRow>> TableViews_ViewModel::getMyStringMultiRowHandlesTableRows()
  {
    return this->MyStringMultiRowHandlesTableRows;
  }
  std::vector<std::string> TableViews_ViewModel::getMyStringMultiRowHandlesTableSelectedRows()
  {
    return this->MyStringMultiRowHandlesTableSelectedRows;
  }
  std::vector<std::shared_ptr<TableViews_ViewModelFirstColumnRowHandleRow>> TableViews_ViewModel::getFirstColumnRowHandleTableRows()
  {
    return this->FirstColumnRowHandleTableRows;
  }
  std::string TableViews_ViewModel::getFirstColumnRowHandleTableSelectedRow()
  {
    return this->FirstColumnRowHandleTableSelectedRow;
  }
  std::vector<std::shared_ptr<TableViews_ViewModelFirstColumnMultiRowHandleRow>> TableViews_ViewModel::getFirstColumnMultiRowHandleTableRows()
  {
    return this->FirstColumnMultiRowHandleTableRows;
  }
  std::vector<std::string> TableViews_ViewModel::getFirstColumnMultiRowHandleTableSelectedRows()
  {
    return this->FirstColumnMultiRowHandleTableSelectedRows;
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TableViews_ViewModel.cs
/// <filename>
///     TableViews_ViewModel.cs
/// </filename>
using System.Collections.Generic;
using widgetassertions;

namespace widgetassertions
{
    public class TableViews_ViewModel
    {
        private System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyElementsRow> MyElementsTableRows = new System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyElementsRow>();

        public virtual System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyElementsRow> getMyElementsTableRows()
        {
            return this.MyElementsTableRows;
        }

        private int MyElementsTableSelectedRow;

        public virtual int getMyElementsTableSelectedRow()
        {
            return this.MyElementsTableSelectedRow;
        }

        private bool isMyElementsTableVisible;

        public virtual bool getIsMyElementsTableVisible()
        {
            return this.isMyElementsTableVisible;
        }

        private bool isMyElementsTableEnabled;

        public virtual bool getIsMyElementsTableEnabled()
        {
            return this.isMyElementsTableEnabled;
        }

        private bool isMyElementsTableDescriptionColumnVisible;

        public virtual bool getIsMyElementsTableDescriptionColumnVisible()
        {
            return this.isMyElementsTableDescriptionColumnVisible;
        }

        private System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyStringRowHandleRow> MyStringRowHandleTableRows = new System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyStringRowHandleRow>();

        public virtual System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyStringRowHandleRow> getMyStringRowHandleTableRows()
        {
            return this.MyStringRowHandleTableRows;
        }

        private string MyStringRowHandleTableSelectedRow;

        public virtual string getMyStringRowHandleTableSelectedRow()
        {
            return this.MyStringRowHandleTableSelectedRow;
        }

        private System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyStringMultiRowHandlesRow> MyStringMultiRowHandlesTableRows = new System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyStringMultiRowHandlesRow>();

        public virtual System.Collections.Generic.List<widgetassertions.TableViews_ViewModelMyStringMultiRowHandlesRow> getMyStringMultiRowHandlesTableRows()
        {
            return this.MyStringMultiRowHandlesTableRows;
        }

        private System.Collections.Generic.List<string> MyStringMultiRowHandlesTableSelectedRows = new System.Collections.Generic.List<string>();

        public virtual System.Collections.Generic.List<string> getMyStringMultiRowHandlesTableSelectedRows()
        {
            return this.MyStringMultiRowHandlesTableSelectedRows;
        }

        private System.Collections.Generic.List<widgetassertions.TableViews_ViewModelFirstColumnRowHandleRow> FirstColumnRowHandleTableRows = new System.Collections.Generic.List<widgetassertions.TableViews_ViewModelFirstColumnRowHandleRow>();

        public virtual System.Collections.Generic.List<widgetassertions.TableViews_ViewModelFirstColumnRowHandleRow> getFirstColumnRowHandleTableRows()
        {
            return this.FirstColumnRowHandleTableRows;
        }

        private string FirstColumnRowHandleTableSelectedRow;

        public virtual string getFirstColumnRowHandleTableSelectedRow()
        {
            return this.FirstColumnRowHandleTableSelectedRow;
        }

        private System.Collections.Generic.List<widgetassertions.TableViews_ViewModelFirstColumnMultiRowHandleRow> FirstColumnMultiRowHandleTableRows = new System.Collections.Generic.List<widgetassertions.TableViews_ViewModelFirstColumnMultiRowHandleRow>();

        public virtual System.Collections.Generic.List<widgetassertions.TableViews_ViewModelFirstColumnMultiRowHandleRow> getFirstColumnMultiRowHandleTableRows()
        {
            return this.FirstColumnMultiRowHandleTableRows;
        }

        private System.Collections.Generic.List<string> FirstColumnMultiRowHandleTableSelectedRows = new System.Collections.Generic.List<string>();

        public virtual System.Collections.Generic.List<string> getFirstColumnMultiRowHandleTableSelectedRows()
        {
            return this.FirstColumnMultiRowHandleTableSelectedRows;
        }
    }
}
